File: javascript/leap/leap.js
class Year {
	constructor(y) {
		this.year = y;
	}
	isLeap(y) {
		return (this.year % 4 === 0 && this.year % 100 !== 0) || this.year % 400 === 0;
	}
}

export default Year;

File: go/leap/leap.go
package leap

// IsLeapYear calculates if given year is a leap year
func IsLeapYear(year int) bool {
	if year%4 != 0 {
		return false
	}
	if year%100 != 0 {
		return true
	}
	if year%400 == 0 {
		return true
	}
	return false
}
